revolsys-core/src/main/java/com/revolsys/record/query/Query.java

```diff
@@ -771,6 +771,17 @@ public Record insertRecord(final Supplier<Record> newRecordSupplier) {
     }
   }
 
+  @Override
+  public boolean isCancelled() {
+    if (super.isCancelled()) {
+      return true;
+    }
+    if (this.cancellable != null && this.cancellable.isCancelled()) {
+      return true;
+    }
+    return false;
+  }
+
```

Comment:
It's weird that BaseObjectWithProperties has the cancelled flag but doesn't implement the cancellable or CancellableProxy interfaces. I guess this is a good way to bridge that.
Do we never cancel queries anywhere else? Somewhat surprising that this hasn't been identified before.